src/onCLS.ts

```diff
@@ -126,6 +173,30 @@ export const onCLS = (onReport: CLSReportCallback, opts?: ReportOpts) => {
           doubleRAF(() => report());
         });
 
+        const reportSoftNavCLS = (entries: SoftNavigationEntry[]) => {
+          entries.forEach((entry) => {
+            if (entry.navigationId) {
+              report(true);
+              metric = initMetric('CLS', 0, 'soft-navigation');
+              metric.pageUrl =
+                performance.getEntriesByType('soft-navigation')[
+                  entry.navigationId - 2
+                ]?.name;
+              report = bindReporter(
+                onReport,
+                metric,
+                thresholds,
+                opts!.reportAllChanges
+              );
+              currentNav = entry.navigationId;
+            }
+          });
+        };
+
+        if (softNavs(opts)) {
+          observe('soft-navigation', reportSoftNavCLS);
```

Comment:
Followup from comment above... I think we should just automatically do this from inside the observe() helper, shared by all metrics

Reply:
Not all metrics need the soft-navigation observer (FCP and FID don’t for example as they can finalise immediately).

For those that do need this to finalise metrics that may not be finalised before then, there are different steps to make so it’s not generic across all the metrics. We could pass in a callback function to handle that, but that’s exactly what this currently does.

Also currently the observe function is for a single observer with a call back on what to do with that. I’m not sure it’s right to change that to also call another observer for some metrics, with a different callback function.

But maybe I’m just not seeing the way to do this? If you have a look at some of the other metrics and still think there’s an easier/cleaner way to do this, then definitely open to it, if you could give me a little nudge as to how you think it could be structure.